spark-jobs/src/main/scala/za/co/absa/enceladus/conformance/HyperConformance.scala

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.conformance
+
+import org.apache.commons.configuration2.Configuration
+import org.apache.spark.sql.functions.{lit, to_date}
+import org.apache.spark.sql.{DataFrame, SparkSession}
+import org.slf4j.{Logger, LoggerFactory}
+import za.co.absa.enceladus.common.Constants._
+import za.co.absa.enceladus.conformance.interpreter.{Always, DynamicInterpreter, FeatureSwitches}
+import za.co.absa.enceladus.dao.MenasDAO
+import za.co.absa.enceladus.dao.menasplugin.{MenasCredentialsFactory, MenasKerberosCredentialsFactory, MenasPlainCredentialsFactory}
+import za.co.absa.enceladus.dao.rest.{MenasConnectionStringParser, RestDaoFactory}
+import za.co.absa.hyperdrive.ingestor.api.transformer.{StreamTransformer, StreamTransformerFactory}
+
+
+class HyperConformance (implicit cmd: ConfCmdConfig,
+                        featureSwitches: FeatureSwitches,
+                        menasBaseUrls: List[String]) extends StreamTransformer {
+  val log: Logger = LoggerFactory.getLogger(this.getClass)
+
+  @throws[IllegalArgumentException]
+  def transform(streamData: DataFrame): DataFrame = {
+    import za.co.absa.enceladus.utils.implicits.DataFrameImplicits.DataFrameEnhancements
+
+    implicit val spark: SparkSession = streamData.sparkSession
+    val menasCredentials = cmd.menasCredentialsFactory.getInstance()
+
+    implicit val dao: MenasDAO = RestDaoFactory.getInstance(menasCredentials, menasBaseUrls)
+    dao.authenticate()
+
+    val reportVersion = getReportVersion
+
+    logPreConformanceInfo(streamData)
+
+    val conformance = dao.getDataset(cmd.datasetName, cmd.datasetVersion)
+
+    DynamicInterpreter.interpret(conformance, streamData)
+      .withColumnIfDoesNotExist(infoDateColumn, to_date(lit(cmd.reportDate), reportDateFormat))
+      .withColumnIfDoesNotExist(infoDateColumnString, lit(cmd.reportDate))
```

Comment:
Good question. As far as I understand Hyperdrive jobs process chunks of data that were accumulated from the previous run in batches crafted by the source system. So the report date might be known for each 'batch'. But I'm not sure.

Reply:
Currently, we don't have any long-running continuous streaming use-cases, but rather once-a-day ingestions that ingest everything that accumulated in the meantime. In such scenarios, a report date can make sense.

What does the report date mean exactly? Is it different from the ingestion date? Since the report date is passed as a config value, does it have to be changed every day?

Reply:
Not completely sure, but here is how I see it.

> What does the report date mean exactly?

For a snapshot-based data, it is the end of the day when the snapshot is made.

> Is it different from the ingestion date?

Yeah, I think it is usually T-1 (where T is the ingestion day).

> Since the report date is passed as a config value, does it have to be changed every day?

Yes, it should be changed each day for data that comes each day. For weekly data it needs to be changed every week, and so on.

Reply:
> Yes, it should be changed each day for data that comes each day. For weekly data it needs to be changed every week, and so on.

Hmm, this might be a problem. Hyperdrive-Trigger doesn't support dynamically changing configuration right now. At the moment it means that the configuration would have to be changed manually every day, which is not a viable solution imo.

Reply:
👍 Interesting, Thanks! Something to think about and discuss.